File: AuctionMarket/src/main/java/com/portfolio/auctionmarket/domain/ratings/repository/RatingRepository.java
package com.portfolio.auctionmarket.domain.ratings.repository;

import com.portfolio.auctionmarket.domain.orders.entity.Order;
import com.portfolio.auctionmarket.domain.ratings.entity.Rating;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.repository.EntityGraph;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

@Repository
public interface RatingRepository extends JpaRepository<Rating, Long> {
    boolean existsByOrder(Order order);

    @EntityGraph(attributePaths = {"toUser", "fromUser", "order"})
    Page<Rating> findAllByToUser_UserId(Long toUserUserId, Pageable pageable);

    Rating findByOrder_OrderId(Long orderId);
}

File: AuctionMarket/auction-domain/src/main/kotlin/org/example/auctioncommon/domain/rating/repository/RatingRepository.kt
package org.example.auctioncommon.domain.rating.repository

import org.example.auctioncommon.domain.order.entity.Order
import org.example.auctioncommon.domain.rating.entity.Rating
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import org.springframework.data.jpa.repository.EntityGraph
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface RatingRepository : JpaRepository<Rating, Long> {
    fun existsByOrder(order: Order): Boolean

    @EntityGraph(attributePaths = ["toUser", "fromUser", "order"])
    fun findAllByToUser_UserId(toUserUserId: Long, pageable: Pageable): Page<Rating>

    fun findByOrder_OrderId(orderId: Long): Rating?
}
